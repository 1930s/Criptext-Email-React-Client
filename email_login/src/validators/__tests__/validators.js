/* eslint-env node, jest */

import {
  validateUsername,
  validateFullname,
  validatePassword,
  validateConfirmPassword,
  validateEmail
} from './../validators';

describe('Validators: ', () => {
  it('Should mark the username as valid (true)', () => {
    const username = 'myusername';
    expect(validateUsername(username)).toMatchSnapshot();
  });

  it('Should mark the username as invalid (false)', () => {
    const username = 'u';
    expect(validateUsername(username)).toMatchSnapshot();
  });

  it('Should mark the fullname as valid (true)', () => {
    const fullname = 'myfullname';
    expect(validateFullname(fullname)).toMatchSnapshot();
  });

  it('Should mark the fullname as invalid (false)', () => {
    const fullname = 'f';
    expect(validateFullname(fullname)).toMatchSnapshot();
  });

  it('Should mark the password as valid (true)', () => {
    const password = '1234';
    expect(validatePassword(password)).toMatchSnapshot();
  });

  it('Should mark the password as invalid (false)', () => {
    const password = 'p';
    expect(validatePassword(password)).toMatchSnapshot();
  });

  it('Should mark the confirmation password as valid (true)', () => {
    const password = '1234';
    const confirmation = '1234';
    expect(validateConfirmPassword(password, confirmation)).toMatchSnapshot();
  });

  it('Should mark the confirmation password as invalid (false)', () => {
    const password = '1234';
    const confirmation = '1235';
    expect(validateConfirmPassword(password, confirmation)).toMatchSnapshot();
  });

  it('Should mark the email as valid (true)', () => {
    const email = 'username@criptext.com';
    expect(validateEmail(email)).toMatchSnapshot();
  });

  it('Should mark the email as invalid (false)', () => {
    const email = 'username.criptext.com';
    expect(validateEmail(email)).toMatchSnapshot();
  });
  it('Should mark the email as invalid (false)', () => {
    const email = '@criptext.com';
    expect(validateEmail(email)).toMatchSnapshot();
  });
  it('Should mark the email as invalid (false)', () => {
    const email = 'username@criptext';
    expect(validateEmail(email)).toMatchSnapshot();
  });
});
