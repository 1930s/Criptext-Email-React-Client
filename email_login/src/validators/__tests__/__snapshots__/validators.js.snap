// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validators:  Should mark the confirmation password as invalid (false) 1`] = `false`;

exports[`Validators:  Should mark the confirmation password as valid (true) 1`] = `true`;

exports[`Validators:  Should mark the email as invalid (false) 1`] = `false`;

exports[`Validators:  Should mark the email as invalid (false) 2`] = `false`;

exports[`Validators:  Should mark the email as invalid (false) 3`] = `false`;

exports[`Validators:  Should mark the email as valid (true) 1`] = `true`;

exports[`Validators:  Should mark the fullname as invalid (false) 1`] = `false`;

exports[`Validators:  Should mark the fullname as valid (true) 1`] = `true`;

exports[`Validators:  Should mark the password as invalid (false) 1`] = `false`;

exports[`Validators:  Should mark the password as valid (true) 1`] = `true`;

exports[`Validators:  Should mark the username as invalid (false) 1`] = `false`;

exports[`Validators:  Should mark the username as valid (true) 1`] = `true`;
